public/views/DataExplorer.vue

```diff
@@ -0,0 +1,376 @@
+          >, {{ selectionNumRows }}
+          <strong class="selected-color">selected</strong>
+        </template>
+      </p>
+    </main>
+
+    <status-sidebar />
+    <status-panel />
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { capitalize, isEmpty } from "lodash";
+
+// Components
+import ActionColumn, { Action } from "../components/layout/ActionColumn.vue";
+import AddVariablePane from "../components/panel/AddVariablePane.vue";
+import DataSize from "../components/buttons/DataSize.vue";
+import FacetListPane from "../components/panel/FacetListPane.vue";
+import FilterBadge from "../components/FilterBadge.vue";
+import LeftSidePanel from "../components/layout/LeftSidePanel.vue";
+import ImageMosaic from "../components/ImageMosaic.vue";
+import SearchInput from "../components/SearchInput.vue";
+import SelectDataTable from "../components/SelectDataTable.vue";
+import SelectGeoPlot from "../components/SelectGeoPlot.vue";
+// import SelectGraphView from "../components/SelectGraphView.vue";
+// import SelectTimeseriesView from "../components/SelectTimeseriesView.vue";
+import StatusPanel from "../components/StatusPanel.vue";
+import StatusSidebar from "../components/StatusSidebar.vue";
+
+// Store
+import { Highlight, RowSelection, Variable } from "../store/dataset/index";
+import { getters as datasetGetters } from "../store/dataset/module";
+import {
+  DATA_EXPLORER_VAR_INSTANCE,
+  ROUTE_PAGE_SUFFIX,
+} from "../store/route/index";
+import { getters as routeGetters } from "../store/route/module";
+import { actions as viewActions } from "../store/view/module";
+
+// Util
+import { overlayRouteEntry } from "../util/routes";
+import { getNumIncludedRows } from "../util/row";
+import { spinnerHTML } from "../util/spinner";
+import { META_TYPES } from "../util/types";
+import {
+  GEO_VIEW,
+  GRAPH_VIEW,
+  IMAGE_VIEW,
+  TABLE_VIEW,
+  TIMESERIES_VIEW,
+  filterViews,
+} from "../util/view";
+
+const ACTIONS = [
+  { name: "All Variables", icon: "database", paneId: "available" },
+  { name: "Text Variables", icon: "font", paneId: "text" },
+  { name: "Categorical Variables", icon: "align-left", paneId: "categorical" },
+  { name: "Number Variables", icon: "bar-chart", paneId: "number" },
+  { name: "Time Variables", icon: "clock-o", paneId: "time" },
+  { name: "Location Variables", icon: "map-o", paneId: "location" },
+  { name: "Image Variables", icon: "image", paneId: "image" },
+  { name: "Unknown Variables", icon: "question", paneId: "unknown" },
+  { name: "Selected Variables", icon: "check", paneId: "selected" },
+  { name: "Create New Variable", icon: "plus", paneId: "add" },
+] as Action[];
+
+export default Vue.extend({
+  name: "DataExplorer",
+
+  components: {
+    ActionColumn,
+    AddVariablePane,
+    DataSize,
+    FacetListPane,
+    LeftSidePanel,
+    ImageMosaic,
+    SearchInput,
+    SelectDataTable,
+    SelectGeoPlot,
+    // SelectGraphView,
```

Comment:
This is for later right?

Reply:
Exactly, these are used on the _Select Training_ page, but commented out, so I added them just to not forget about their implementation.